src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,123 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int BIT_SHIFT_VALUE = 1;
+    private static final int ONE_ELEMENT_SIZE = 1;
+    private static final int EMPTY_LIST_VALUE = 0;
+    private static final int FIRST_ELEMENT_INDEX = 0;
+    private static final int NO_SUCH_ELEMENT_INDEX = -1;
+    private static final String NO_SUCH_ELEMENT_MESSAGE = "There is no such element in list.";
+    private static final String INDEX_OUT_OF_BOUNDS_MESSAGE = "This index does not exist in list.";
+
+    private Object[] data;
+    private int size;
+
+    public ArrayList() {
+        data = new Object[DEFAULT_CAPACITY];
+        size = EMPTY_LIST_VALUE;
```

Comment:
I don't know how necessary it is to initialize the primitive in the constructor, let the mentor decide

Reply:
Yes, according to [this](https://mate-academy.github.io/jv-program-common-mistakes/java-core/generics/storage.html#all-non-constant-fields-should-be-initialized-in-the-constructor)

Reply:
@proxychang Well, I believe that is the case, when we are initialising our instance variables with some non-default values. Here, it is absolutely logical per `ArrayList` creation to have default(0) `size` value. Although it is not a mistake, that could be a bit confusing to see we are doing `size` non-default initialisation.